pandas/tests/series/indexing/test_where.py

```diff
@@ -452,3 +452,18 @@ def test_where_empty_series_and_empty_cond_having_non_bool_dtypes():
     ser = Series([], dtype=float)
     result = ser.where([])
     tm.assert_series_equal(result, ser)
+
+
+def test_where_category_not_dropping():
+
+    s = pd.Series(["A", "A", "B", "B", "C"], dtype="category")
+    exp = s.dtype
+    s.where(s != "C", s)
+    res = s.dtype
+    assert exp == res
```

Comment:
you want to construct the expected result and test for the equality of the series using `tm.assert_series_equal` - see neighboring tests for examples

Reply:
use 
``` python
res = ...
expected = ...
assert_..._equal(res, expected)
```
in both tests